File: fabric/common/src/main/java/com/noxcrew/noxesium/fabric/network/clientbound/ClientboundCustomSoundStopPacket.java
package com.noxcrew.noxesium.fabric.network.clientbound;

import com.noxcrew.noxesium.api.fabric.network.NoxesiumPacket;
import com.noxcrew.noxesium.api.fabric.network.NoxesiumPayloadType;
import com.noxcrew.noxesium.fabric.network.CommonPackets;
import net.minecraft.network.RegistryFriendlyByteBuf;
import net.minecraft.network.codec.StreamCodec;
import net.minecraft.network.protocol.common.custom.CustomPacketPayload;

/**
 * Sent by a server to stop a custom Noxesium sound by its id.
 */
public record ClientboundCustomSoundStopPacket(int id) implements NoxesiumPacket {
    public static final StreamCodec<RegistryFriendlyByteBuf, ClientboundCustomSoundStopPacket> STREAM_CODEC =
            CustomPacketPayload.codec(ClientboundCustomSoundStopPacket::write, ClientboundCustomSoundStopPacket::new);

    private ClientboundCustomSoundStopPacket(RegistryFriendlyByteBuf buf) {
        this(buf.readVarInt());
    }

    private void write(RegistryFriendlyByteBuf buf) {
        buf.writeVarInt(id);
    }

    @Override
    public NoxesiumPayloadType<?> noxesiumType() {
        return CommonPackets.INSTANCE.CLIENT_CUSTOM_SOUND_STOP;
    }
}

File: paper/legacy/src/main/kotlin/com/noxcrew/noxesium/legacy/api/network/clientbound/ClientboundCustomSoundStopPacket.kt
package com.noxcrew.noxesium.legacy.api.network.clientbound

import com.noxcrew.noxesium.paper.api.network.NoxesiumPacket
import com.noxcrew.noxesium.paper.api.network.NoxesiumPackets

/**
 * Sent by a server to stop a custom Noxesium sound by its id.
 */
public data class ClientboundCustomSoundStopPacket(
    public val id: Int,
) : NoxesiumPacket(NoxesiumPackets.CLIENT_STOP_SOUND)
